dump/authorization.py

```diff
@@ -0,0 +1,122 @@
+# [START installed_apps]
+
+import httplib2
+import pprint
+import sys
+
+from apiclient.discovery import build
+from apiclient.errors import HttpError
+
+from oauth2client.client import OAuth2WebServerFlow
+from oauth2client.client import AccessTokenRefreshError
+from oauth2client.client import flow_from_clientsecrets
+from oauth2client.file import Storage
+from oauth2client.tools import run
+
+# Enter your Google Developer Project number
+PROJECT_NUMBER = 'XXXXXXXXXXXXX'
+FLOW = flow_from_clientsecrets('client_secrets.json',
+                               scope='https://www.googleapis.com/auth/bigquery')
+
+
+def main():
+  storage = Storage('bigquery_credentials.dat')
+  credentials = storage.get()
+
+  if credentials is None or credentials.invalid:
+    credentials = run(FLOW, storage)
+
+  http = httplib2.Http()
+  http = credentials.authorize(http)
+
+  bigquery_service = build('bigquery', 'v2', http=http)
+
+  try:
+    datasets = bigquery_service.datasets()
+    listReply = datasets.list(projectId=PROJECT_NUMBER).execute()
+    print 'Dataset list:'
+    pprint.pprint(listReply)
+
+  except HttpError as err:
+    print 'Error in listDatasets:', pprint.pprint(err.content)
+
+  except AccessTokenRefreshError:
+    print ("Credentials have been revoked or expired, please re-run"
+           "the application to re-authorize")
+
+if __name__ == '__main__':
+  main()
+
+# [END installed_apps]
+
+
+# [START service_accounts]
```

Comment:
Please consider using the new default credentials:
https://developers.google.com/accounts/docs/application-default-credentials

Talk to @anthmgoogle for more details.